hivemq-edge/src/main/java/com/hivemq/api/resources/impl/SamplingResourceImpl.java

```diff
@@ -0,0 +1,103 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hivemq.api.model.samples.PayloadSample;
+import com.hivemq.api.model.samples.PayloadSampleList;
+import com.hivemq.api.resources.SamplingApi;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.sampling.SamplingService;
+import com.hivemq.util.ErrorResponseUtil;
+import com.saasquatch.jsonschemainferrer.AdditionalPropertiesPolicies;
+import com.saasquatch.jsonschemainferrer.JsonSchemaInferrer;
+import com.saasquatch.jsonschemainferrer.RequiredPolicies;
+import com.saasquatch.jsonschemainferrer.SpecVersion;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+
+@Singleton
+public class SamplingResourceImpl implements SamplingApi {
+
+    private final @NotNull SamplingService samplingService;
+    private final @NotNull ObjectMapper objectMapper;
+    private static final JsonSchemaInferrer INFERRER = JsonSchemaInferrer.newBuilder()
+            .setSpecVersion(SpecVersion.DRAFT_07)
+            .setAdditionalPropertiesPolicy(AdditionalPropertiesPolicies.notAllowed())
+            .setRequiredPolicy(RequiredPolicies.nonNullCommonFields())
+            .build();
+
+
+    @Inject
+    public SamplingResourceImpl(
+            final @NotNull SamplingService samplingService, final @NotNull ObjectMapper objectMapper) {
+        this.samplingService = samplingService;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public @NotNull Response getSamplesForTopic(@NotNull final String topicBase64) {
+
+        final String topic = new String(Base64.getDecoder().decode(topicBase64), StandardCharsets.UTF_8);
+
+        final List<byte[]> samples = samplingService.getSamples(topic);
+        final ArrayList<PayloadSample> sampleArrayList = new ArrayList<>();
+
+        // we want LIFO, but the queue return FIFO
+        Collections.reverse(samples);
+
+        samples.forEach(sample -> sampleArrayList.add(new PayloadSample(Base64.getEncoder().encodeToString(sample))));
+        return Response.ok().entity(new PayloadSampleList(sampleArrayList)).build();
+    }
+
+    @Override
+    public @NotNull Response getSchemaForTopic(@NotNull final String topicBase64) {
+        final String topic = new String(Base64.getDecoder().decode(topicBase64), StandardCharsets.UTF_8);
+        final List<byte[]> samples = samplingService.getSamples(topic);
+        if (samples.isEmpty()) {
+            return ErrorResponseUtil.notFound("samples", "topic");
+        }
+
+        final ArrayList<JsonNode> jsonSamples = new ArrayList<>();
+        for (final byte[] sample : samples) {
+            try {
+                jsonSamples.add(objectMapper.readTree(sample));
+            } catch (final IOException e) {
+                // TODO
```

Comment:
Maybe some logs here?

Reply:
question is whether we should log the broken payload here. Guess the Exception message must be enough.